src/pages/Admin/Venue/Users/VenueUsersReport.tsx

```diff
@@ -0,0 +1,166 @@
+import React, { useEffect, useCallback, useState } from "react";
+import { useWorldUsers } from "hooks/users";
+import "./VenueUsersReport.scss";
+import Table from "react-bootstrap/Table";
+import Form from "react-bootstrap/Form";
+import { downloadGeneratedCSVFile } from "utils/csv";
+import { User } from "types/User";
+import { WithId } from "utils/id";
+
+const VenueUsersReport: React.FC = () => {
+  const { worldUsers } = useWorldUsers();
+
+  const [headers, setHeaders] = useState<Array<string>>([]);
+  const [headersSelected, setHeadersSelected] = useState<Array<string>>([]);
+  const [customFieldValue, setCustomFieldValue] = useState("");
+
+  const userIsComplete = useCallback((user: WithId<User>) => {
+    return user.partyName && user.pictureUrl;
+  }, []);
+
+  const getUsersHeaders = useCallback(() => {
+    let index = 0;
+    while (index < worldUsers.length) {
+      if (userIsComplete(worldUsers[index])) {
+        setHeaders(Object.keys(worldUsers[index]));
+        return;
+      } else {
+        index++;
+      }
+    }
+  }, [worldUsers, userIsComplete]);
+
+  useEffect(() => {
+    getUsersHeaders();
+  }, [worldUsers, getUsersHeaders]);
+
+  const renderPageHeader = () => {
+    return <h4>Users Reports</h4>;
+  };
```

Comment:
We don't like to use these 'render functions'. For something this small it probably should just be done inline.

For bigger things (eg. `renderUserTable` below), it should most likely be a sub-component instead.

Reply:
fixed in 12e5c07fea64a78aced31d706faf59c5e523c733